### ISSUE TITLE
Transfer to clipboard manager does not work on KDE

### ISSUE THREAD

**Issue by ad8e** [glfw/glfw#2270]
GLFW version: 3.3.8-1 (from Ubuntu packages)
X11

It doesn't replace the clipboard contents, and doesn't return an error.

Testcase:
Download [clipboard.txt](https://github.com/glfw/glfw/files/10678855/clipboard.txt), rename it to clipboard.c, compile with "gcc clipboard.c -lglfw"
Copy "text" to clipboard with Ctrl+C.
./a.out will output in console: Setting clipboard to "Hello GLFW World!"
But the clipboard does not change, it is still "text".

glfwGetClipboardString still works.

**Comment by elmindreda** [comment]
Oh, yes, this is because of how the clipboard works on X11.

**TL;DR** You need to process events for this to work consistently.

On X11 the application doesn't upload the new clipboard contents to the window system. Instead it becomes the owner of the clipboard and other applications can request the contents on demand over an asynchronous protocol. Since the program above doesn't process events this process times out and fails. During `glfwTerminate` GLFW tries to hand the contents over to the clipboard manager, but if one isn't running that fails too.

**Comment by ad8e** [comment]
My Kubuntu X11 system comes with the clipboard manager Klipper by default, and I left it running. I guess this clipboard manager does not capture the handoff during glfwTerminate().

**Comment by elmindreda** [comment]
You're right, that seems to be a bug in GLFW on KDE. I can see it trying to hand the contents over to the clipboard manager but it fails somehow.

**Comment by ad8e** [comment]
SDL2 has the same problem: something fed to `SDL_SetClipboardText` does not appear when pressing Ctrl+V.

If you call `glfwSetClipboardString`, then `glfwGetClipboardString()`, the Get call receives the string from Set. So the program can see its own clipboard, but other programs can't. SDL2 has the same behavior here as well.

Thus, it's likely a Klipper problem, not a GLFW problem.
